packages/medusa/src/services/product-category.ts

```diff
@@ -65,6 +65,38 @@ class ProductCategoryService extends TransactionBaseService {
 
     return productCategoryTree
   }
+
+  /**
+   * Deletes a product category
+   *
+   * @param productCategoryId is the id of the product category to delete
+   * @return a promise
+   */
+  async delete(productCategoryId: string): Promise<void> {
+    return await this.atomicPhase_(async (manager) => {
+      const productCategoryRepository: ProductCategoryRepository =
+        manager.getCustomRepository(this.productCategoryRepo_)
+
+      const productCategory = await this.retrieve(productCategoryId, {
+        relations: ["category_children"],
+      })
+
+      if (!productCategory) {
+        return Promise.resolve()
+      }
+
+      if (productCategory.category_children.length > 0) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_ALLOWED,
+          `Deleting ProductCategory (${productCategoryId}) with category children is not allowed`
+        )
+      }
+
+      await productCategoryRepository.softRemove(productCategory)
```

Comment:
comment(non-blocking): Bringing the discussion down here @adrien2p, we should be able to hard-delete categories, as we don't benefit from any of the usual advantages that come from soft-deletion

Reply:
moved this to hard delete.